src/Widgets/DataFormUploadWidget/DataFormUploadWidgetComponent.tsx

```diff
@@ -1,12 +1,38 @@
-import { ContentTag, InPlaceEditingOff, provideComponent } from 'scrivito'
+import {
+  ContentTag,
+  currentLanguage,
+  InPlaceEditingOff,
+  provideComponent,
+} from 'scrivito'
 import { DataFormUploadWidget } from './DataFormUploadWidgetClass'
 import { OverlayTrigger, Popover } from 'react-bootstrap'
+import { useDropzone } from 'react-dropzone'
+import { Attachment } from '../../Components/Attachment'
+import { useEffect } from 'react'
 
 provideComponent(DataFormUploadWidget, ({ widget }) => {
   const id = ['DataFormUploadWidget', widget.id()].join('-')
-
   const attributeName = widget.get('attributeName')
 
+  const { acceptedFiles, getRootProps, getInputProps, inputRef } = useDropzone()
+
+  useEffect(() => {
+    if (!inputRef.current) return
+
+    const dataTransfer = new DataTransfer()
+    acceptedFiles.forEach((file) => dataTransfer.items.add(file))
+
+    inputRef.current.files = dataTransfer.files
+  }, [acceptedFiles, inputRef])
```

Comment:
* [ ] The `inputRef` dependency is misleading. This is the 2nd case described [here](https://www.epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array#custom-hooks-and-the-ref-object-itself). I didn't read the full article and can't comment on a working solution but the current code could lead to confusion if devs start copy/pasting it. If there is no clean solution maybe a comment would help?

Reply:
I would argue, that this is more the 3rd case described in the same section, since the `inputRef` comes from `useDropzone`. At least eslint is complaining:

<img width="1017" alt="Screenshot 2024-09-04 at 11 08 01" src="https://github.com/user-attachments/assets/8be1e15e-ced2-4c87-b90d-db7977fd5426">

To quote from the same article:

> The reason we get the warning there when we didn't get it within the component is because ESLint is pretty limited in its ability to trace what you're doing with your JavaScript. So the React plugin for ESLint can't know that cbRef is actually a ref. So just to be safe, it warns you.
> 
> Luckily, as we just learned, including it in the dependency array doesn't make any difference anyway, so just include it!

I would just leave it as is.